src/screens/home.tsx

```diff
@@ -1,17 +1,162 @@
-import { StyleSheet, View } from 'react-native'
+import React, { Dispatch, SetStateAction, useState, useEffect } from 'react'
+import { Linking, ScrollView, StyleSheet, View, Image } from 'react-native'
 import { Button, Text } from 'react-native-paper'
+import DropDown from 'react-native-paper-dropdown'
+import { getStopData } from '../clients/APICalls'
+import StopsDropDown from '../components/dropDownStops'
 import { ScreenNavigationProps } from '../routes'
+import DetailsScreen from './details'
+import tram1 from './tram1.png'
+import tram2 from './tram2.png'
+import refreshButton from './refreshButton.png'
 
 type HomeScreenProps = ScreenNavigationProps<'Home'>
 
-const HomeScreen = ({ navigation }: HomeScreenProps) => (
-  <View style={styles.container}>
-    <Text style={styles.text}>Home Screen</Text>
-    <Button mode="contained" onPress={() => navigation.navigate('Details')}>
-      Go to details
-    </Button>
-  </View>
-)
+export type Departure = {
+  destination: string
+  waitTime: number
+  platformNumber: number
+  isSingle: boolean
+}
+export type Station = {
+  value: Board[]
+}
+
+type Board = {
+  Wait0: string
+  Dest0: string
+  Carriages0: string
+  Wait1: string
+  Dest1: string
+  Carriages1: string
+  Wait2: string
+  Dest2: string
+  Carriages2: string
+  Wait3: string
+  Dest3: string
+  Carriages3: string
+  AtcoCode: string
+}
+const HomeScreen = ({ navigation }: HomeScreenProps) => {
+  const [details, setDetails] = useState<Departure[]>([])
+  const [dropDownVisible, setDropDownVisible] = useState(true)
+  const [station, setStation] = useState('')
+  const [pageContent, setPageContent] = useState(<></>)
```

Comment:
Hmm I'm not sure why, but having components in states is unconventional. Just have it as a normal variable that depends on other states/props